src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs

```diff
@@ -52,6 +52,7 @@ internal static class SymbolExtensions
 
     private static readonly HashSet<string> s_supportedSystemTypes = new()
     {
+        "System.DateTimeKind",
```

Comment:
Lets add `datetimeoffset` and a test.

Reply:
Added Tests to Existing Tests(reference BuildersBsonAttributes.cs and LinqBsonAttributes.cs)